### PR TITLE
Test framework and example for modes

### PR DESCRIPTION
**Pull request by bhourahine**
Only runs for serial code, as modes is not mpi parallel.

### CODE REVIEW

File: test/prog/modes/C24O6H8/dftb_in.hsd [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+# This file uses the 5th input format
+ParserOptions = {
+  ParserVersion = 7
+}
+
+# if run in MPI parallel, allow openMP threads to be used
+Parallel = {
+  UseOmpThreads = Yes
+}
```
Comment (awvwgk) [inline test/prog/modes/C24O6H8/dftb_in.hsd:39]: For what reason is a `dftb+` input file needed for `modes`?

File: test/prog/modes/README [hunk 2]
```diff
@@ -0,0 +1,92 @@
+AUTOTEST
+========
```
Comment (awvwgk) [inline test/prog/modes/README:2]: How about converting this one to reStructuredText, even if it is only for the nice preview on GH?

File: test/prog/modes/C24O6H8/dftb_in.hsd (hunk 1 above)
Comment (bhourahine) [reply test/prog/modes/C24O6H8/dftb_in.hsd:39]: It isn't (parent file for hessian generation), will remove

File: test/prog/modes/README.rst [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+The autotest suite contains some tests for the modes code from the DFTB+
+project, with precalculated results for automated comparison. In order to test
+this binary on your system, you should execute the following command:
+
+  ./bin/autotest2 \
+     -w <tempdir> \
+     -p <binary> \
+     -f <testfile> \
+     -s A \
+     -v
```
Comment (awvwgk) [inline test/prog/modes/README.rst:17]:
```suggestion
.. code:: bash

  ./bin/autotest2 \
     -w <tempdir> \
     -p <binary> \
     -f <testfile> \
     -s A \
     -v
```

File: test/prog/modes/README.rst [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+where
+
+  <tempdir> is an existing directory which should contain the results of the
+             calculations to be validated (this directory is not deleted
+             afterwards)
+
+  <binary> is the path to your binary
+
+  <testfile> contains the list of tests which should be calculated (you can use
+             the 'tests' file in the this directory or create your own file)
```
Comment (awvwgk) [inline test/prog/modes/README.rst:28]:
```suggestion
tempdir
             is an existing directory which should contain the results of the
             calculations to be validated (this directory is not deleted
             afterwards)

binary
             is the path to your binary

testfile
             contains the list of tests which should be calculated (you can use
             the 'tests' file in the this directory or create your own file)
```

File: test/prog/modes/README (hunk 2 above)
Comment (bhourahine) [reply test/prog/modes/README]: Done. Of course it means the DFTB+ test README also needs a PR.

**Comment by bhourahine** [comment]
> Maybe change the first block to code and the second to description. My `rst` is not that good to trust in it blindly, better look at the preview before accepting my suggested changes.

Thanks for the rst suggestions, seems to work.

**Pull request merged**